types/common.ts

```diff
@@ -29,9 +29,13 @@ export class ArrayCursor<T> {
     return this.idx < this.data.length;
   }
 
-  public pollNext() {
-    if (!this.hasNext()) return undefined;
-    return this.data[this.idx++];
+  public async pollNext() {
+    while (this.hasNext()) {
+      const res = this.data[this.idx++];
+      await res.getData();
+      if (res.exists) return res;
```

Comment:
should we go ahead and delete the stale refs here instead of just skipping them?

Reply:
Talked in person, no. Landing to avoid headache after The Refactor